include/pm/device.h

```diff
@@ -407,6 +410,44 @@ bool pm_device_wakeup_is_enabled(const struct device *dev);
  * @retval false If the device is not wake up capable.
  */
 bool pm_device_wakeup_is_capable(const struct device *dev);
+
+/**
+ * @brief Lock current device state.
+ *
+ * This function locks the current device state. After the state be
+ * locked the system power management and device runtime power
+ * management will not longer change the device state until the state
+ * be unlocked.
```

Comment:
A little rewording makes this clearer possibly
"This function locks the current device power state. Once locked the device state
will not be change by system power management or device runtime power management until unlocked."

Reply:
Indeed, much better with the change.